virtcontainers/tap_endpoint.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2018 Huawei Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/containernetworking/plugins/pkg/ns"
+
+	"github.com/vishvananda/netlink"
+
+	"github.com/kata-containers/runtime/virtcontainers/pkg/uuid"
+)
+
+// TapEndpoint represents just a tap endpoint
+type TapEndpoint struct {
+	ID                 string
+	NicName            string
+	TAPIface           NetworkInterface
+	VMFds              []*os.File
```

Comment:
Is this needed in case of `tap`? I thought only macvtap was requiring this.
/cc @amshinde

Reply:
When using qemu command line/qmp to insert a nic, We can use both name and fds. With this filed, tap/veth/macvtap can be hotpluged by the same way. Actually veth endpoint has this field too.